python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -0,0 +1,594 @@
+
+
+class MatchPessimisticBConfig(pexConfig.Config):
+    """Configuration for MatchPessimisticBTask
+    """
+    numBrightStars = pexConfig.RangeField(
+        doc="Number of bright stars to use. Sets the max number of patterns "
+            "that can be tested.",
+        dtype=int,
+        default=200,
+        min=2,
+    )
+    minMatchedPairs = pexConfig.RangeField(
+        doc="Minimum number of matched pairs; see also minFracMatchedPairs.",
+        dtype=int,
+        default=30,
+        min=2,
+    )
+    minFracMatchedPairs = pexConfig.RangeField(
+        doc="Minimum number of matched pairs as a fraction of the smaller of "
+            "the number of reference stars or the number of good sources; "
+            "the actual minimum is the smaller of this value or "
+            "minMatchedPairs.",
+        dtype=float,
+        default=0.3,
+        min=0,
+        max=1,
+    )
+    matcherIterations = pexConfig.RangeField(
+        doc="Number of softening interations in matcher.",
+        dtype=int,
+        default=5,
+        min=1,
+    )
+    maxOffsetPix = pexConfig.RangeField(
+        doc="Maximum allowed shift of WCS, due to matching (pixel). "
+            "MatchPessimisticBTask converts this to arseconds internally "
+            "using the wcs pixel scale.",
+        dtype=int,
+        default=300,
+        max=4000,
+    )
+    maxRotationDeg = pexConfig.RangeField(
+        doc="Rotation angle allowed between sources and position reference "
+            "objects (degrees).",
+        dtype=float,
+        default=1.0,
+        max=6.0,
+    )
+    numPointsForShape = pexConfig.Field(
+        doc="Number of points to define a shape for matching.",
+        dtype=int,
+        default=6,
+    )
+    numPointsForShapeAttempt = pexConfig.Field(
+        doc="Number of points to try for creating a shape. This value should "
+            "be greater than numPointsForShape.",
+        dtype=int,
+        default=7,
+    )
+    numPatternConsensus = pexConfig.Field(
+        doc="Number of implied shift/rotations from patterns that must agree "
+            "before it a given shift/rotation is accepted. This is only used "
+            "after the first softening iteration fails and if both the "
+            "number of reference and source objects is greater than "
+            "numBrightStars.",
+        dtype=int,
+        default=3,
+    )
+    sourceSelector = sourceSelectorRegistry.makeField(
+        doc="How to select sources for cross-matching. The default "
+            "matcherSourceSelector removes objects with low S/N, bad "
+            "saturated objects, edge objects, and interpolated objects.",
+        default="matcher"
+    )
+
+    def setDefaults(self):
+        sourceSelector = self.sourceSelector["matcher"]
+        sourceSelector.setDefaults()
+
+    # The paremeters below are not used in this matcher and are kept around
+    # for compatability.
```

Comment:
Here's the problem with removing these through. Right now the processCcd.py config for obs_subaru/hsc sets these parameters by hand. Removing them here would cause breaks for the pessimistic task that would be difficult to track down. Once we do the RFC to replace the current matcher I'm happy to remove them and get rid of the defaults in obs_subaru but for now to keep things as they are I'm okay with having them stick around considering we will have an upcoming ticket/epic to change to the Pessimistic matcher.